### PR TITLE
curvefs/client: aync upload write disk file

### PR DESCRIPTION
**Pull request by wuhongsong**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?): 

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: curvefs/src/client/s3/disk_cache_write.cpp [hunk 1]
```diff
@@ -100,13 +102,18 @@ int DiskCacheWrite::UploadFile(const std::string name) {
...
         return -1;
     }
-    ret = client_->Upload(name, buffer, fileSize);
-    if (ret < 0) {
-        LOG(ERROR) << "upload object fail. object: " << name;
-        posixWrapper_->free(buffer);
-        posixWrapper_->close(fd);
-        return -1;
-    }
+    PutObjectAsyncCallBack cb =
```
Comment (ilixiaocui) [inline curvefs/src/client/s3/disk_cache_write.cpp:105]: This definition in many places，make it public

File: curvefs/src/client/s3/disk_cache_write.cpp [hunk 2]
```diff
@@ -139,7 +146,7 @@ int DiskCacheWrite::AsyncUploadFunc() {
         std::list<std::string>::iterator iter;
         int ret;
         for (iter = toUpload.begin(); iter != toUpload.end(); iter++) {
-            ret = UploadAndRemove(*iter);
+            ret = UploadFile(*iter);
```
Comment (ilixiaocui) [inline curvefs/src/client/s3/disk_cache_write.cpp:149]: Why not remove? When will it be deleted？

Comment (wuhongsong) [reply curvefs/src/client/s3/disk_cache_write.cpp:149]: > Why not remove? When will it be deleted？

delete in aync callback.

File: curvefs/src/client/s3/disk_cache_write.cpp [hunk 3]
```diff
@@ -100,13 +102,18 @@ int DiskCacheWrite::UploadFile(const std::string name) {
...
-    ret = client_->Upload(name, buffer, fileSize);
-    if (ret < 0) {
-        LOG(ERROR) << "upload object fail. object: " << name;
-        posixWrapper_->free(buffer);
-        posixWrapper_->close(fd);
-        return -1;
-    }
+    PutObjectAsyncCallBack cb =
+        [&](const std::shared_ptr<PutObjectAsyncContext> &context) {
+            RemoveFile(context->key);
```
Comment (wu-hanqing) [inline curvefs/src/client/s3/disk_cache_write.cpp:107]: `RemoveFiles`'s result is ignored?

**Review by wu-hanqing — APPROVED** [review]

**Review by ilixiaocui — APPROVED** [review]

File: curvefs/src/client/s3/disk_cache_write.cpp (hunk 1 above)
Comment (wuhongsong) [reply curvefs/src/client/s3/disk_cache_write.cpp:105]: The callback content is not the same

**Pull request merged**